### PR TITLE
Update versions of common packages used in tests and examples

### PR DESCRIPTION
**Pull request by utpilla**
## Changes
- Update versions of common packages used in tests and examples
- Update Wcf example project

### CODE REVIEW

File: examples/wcf/client-core/Examples.Wcf.Client.Core.csproj [hunk 1]
```diff
@@ -1,16 +1,16 @@
-﻿<Project Sdk="Microsoft.NET.Sdk">
+<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
     <OutputType>Exe</OutputType>
-    <TargetFramework>netcoreapp3.1</TargetFramework>
```
Comment (Kielek) [inline examples/wcf/client-core/Examples.Wcf.Client.Core.csproj:5]: Based on file name, here you should have net6.0/net7.0. .NET Framework is handled by examples/wcf/client-netframework/Examples.Wcf.Client.NetFramework.csproj

File: build/Common.nonprod.props [hunk 2]
```diff
@@ -20,11 +20,11 @@
...
       Refer to https://docs.microsoft.com/en-us/nuget/concepts/package-versioning for semver syntax.
     -->
-    <BenchmarkDotNetPkgVer>[0.13.2,0.14)</BenchmarkDotNetPkgVer>
-    <CoverletCollectorPkgVer>[3.1.2,4.0.0)</CoverletCollectorPkgVer>
+    <BenchmarkDotNetPkgVer>[0.13.3,0.14)</BenchmarkDotNetPkgVer>
+    <CoverletCollectorPkgVer>[3.2.0,4.0.0)</CoverletCollectorPkgVer>
     <DotNetXUnitCliVer>[2.3.1,3.0)</DotNetXUnitCliVer>
     <MicrosoftExtensionsLoggingPkgVer>[5.0.0,7.0)</MicrosoftExtensionsLoggingPkgVer>
-    <MicrosoftNETTestSdkPkgVer>[16.11.0,17.0)</MicrosoftNETTestSdkPkgVer>
+    <MicrosoftNETTestSdkPkgVer>[17.3.2,18.0)</MicrosoftNETTestSdkPkgVer>
```
Comment (Kielek) [inline build/Common.nonprod.props:27]: Is there any reason to not reference [17.4.1](https://www.nuget.org/packages/Microsoft.NET.Test.Sdk/17.4.1)?

File: examples/grpc.core/Examples.GrpcCore.AspNetCore/Examples.GrpcCore.AspNetCore.csproj [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
   <PropertyGroup>
     <TargetFramework>net6.0</TargetFramework>
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="OpenTelemetry.Exporter.Console" Version="1.0.1" />
-    <PackageReference Include="OpenTelemetry.Extensions.Hosting" Version="1.0.0-rc2" />
-    <PackageReference Include="OpenTelemetry.Instrumentation.AspNetCore" Version="1.0.0-rc2" />
+    <PackageReference Include="OpenTelemetry.Exporter.Console" Version="$(OpenTelemetryCoreLatestVersion)" />
+    <PackageReference Include="OpenTelemetry.Extensions.Hosting" Version="1.0.0-rc9.4" />
```
Comment (Kielek) [inline examples/grpc.core/Examples.GrpcCore.AspNetCore/Examples.GrpcCore.AspNetCore.csproj:9]: Great to see updated examples.
I have some doubts about using configuration variables `OpenTelemetryCoreLatestVersion`.
It is additional layer to understand by endusers.

File: examples/wcf/client-core/Examples.Wcf.Client.Core.csproj (hunk 1 above)
Comment (utpilla) [reply examples/wcf/client-core/Examples.Wcf.Client.Core.csproj:5]: Good catch! Now that `netcoreapp3.1` is not supported, I think we could rename this project to `Examples.Wcf.Client.DotNet` and have `net6.0` as the TargetFramework. What do you think?

File: build/Common.nonprod.props (hunk 2 above)
Comment (utpilla) [reply build/Common.nonprod.props:27]: We still have some test projects that run on net452. The package version 17.4.1 is not supported on net452.

The package version 17.3.2 is the latest stable version that is supported on net452.

File: examples/grpc.core/Examples.GrpcCore.AspNetCore/Examples.GrpcCore.AspNetCore.csproj (hunk 3 above)
Comment (utpilla) [reply examples/grpc.core/Examples.GrpcCore.AspNetCore/Examples.GrpcCore.AspNetCore.csproj:9]: OpenTelemetry main repo does not have the package versions explicitly mentioned in the .csproj of example projects either.  Ideally, these things should be mentioned in the README.md for these projects.

I don't have a strong opinion here. I can change these back to the specific versions. Let me know.

Comment (Kielek) [reply examples/grpc.core/Examples.GrpcCore.AspNetCore/Examples.GrpcCore.AspNetCore.csproj:9]: It is easier to maintain current version. Lets keep it as it is.

File: examples/wcf/client-core/Examples.Wcf.Client.Core.csproj (hunk 1 above)
Comment (utpilla) [reply examples/wcf/client-core/Examples.Wcf.Client.Core.csproj:5]: I have updated it.

**Review by Kielek — APPROVED** [review]

**Pull request merged**